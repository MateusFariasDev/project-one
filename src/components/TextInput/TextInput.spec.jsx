import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { TextInput } from '.';

describe('<TextInput />', () => {
  it('should have the logo', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue={''} />);

    expect(screen.getByText(/coda/i)).toBeInTheDocument();
    expect(screen.getByText(/dev/i)).toBeInTheDocument();
  });
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue={'testando'} />);

    const input = screen.getByPlaceholderText(/type your search/i);
    expect(input.value).toBe('testando');
  });
  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue={'some value'} />);

    const input = screen.getByPlaceholderText(/type your search/i);
    const value = 'value';

    userEvent.type(input, value);

    expect(input.value).toBe('some value');
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleChange={fn} searchValue={''} />);

    expect(container).toMatchSnapshot();
  });
});
