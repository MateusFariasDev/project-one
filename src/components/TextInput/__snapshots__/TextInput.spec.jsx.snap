// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="logo"
    >
      Coda
      <span>
        Dev
      </span>
    </div>
    <div
      class="search-bar"
    >
      <input
        class="text-input"
        placeholder="Type your search"
        type="search"
        value=""
      />
    </div>
  </div>
  <hr />
</div>
`;
